### ISSUE TITLE
SetTimeout necessary to change ActionBar

### ISSUE THREAD

**Issue by Lelelo1** [NativeScript/NativeScript#7640]
Note only tested on iOS

**Environment**
 - CLI: 6.0.2
 - iOS Runtime: 6.0.1

**Describe the bug**
UiNavigationBar can't be modified in `ActionBar.loadedEvent` event handler.

**To Reproduce**
Create a new project (tested with react-nativescript and plain TypeScript project - hello world starter template). Set an id for ActionBar and add an event handler to it - *in the page navigatingToFunction code behind...*

```
    const actionBar = page.getViewById("actionBar") as ActionBar;
    actionBar.addEventListener(ActionBar.loadedEvent, (event) => {
        // (actionBar.ios as UINavigationBar).topItem.title = "I wwaaas changed"; // does not work
        setTimeout(() => {
            (actionBar.ios as UINavigationBar).topItem.title = "I wwaaas changed"; // works
        }, 0.000000001);
    });
```

 
**Expected behavior**
The changes to the ActionBar on iOS - UINavigationBar should take effect following `ActionBar.loadedEvent`.

**Additional context**
I have tested changing a plenty UINavigationBar values in react-nativescript and only the `topitem.title` in plain typescript project. Have not tested on Android.

**Comment by Lelelo1** [comment]
I also need a timeout when changing `ÙIViewController` (only tested in react-nativescript).

**Comment by NickIliev** [comment]
@Lelelo1 this seems expected as `ActionBar` on `iOS` is actually `UINavigaitonBar` and you are using `navigatingTo` event. So using a setTimeout even with the smallest possible amount is triggered on the next pass of the VM ()and will be probably executed after the navigation event.

To prevent the usage of `setTImeout` try using the reference of the ActionBar one it is loaded (using the `loaded` event instead of a navigation-related event).

e.g.,
```XML
<ActionBar loaded="onActionBarLoaded"></ActionBar>
```

TypeScript
```TypeScript
export function onActionBarLoaded(args: EventData) {
    const ab = args.object as ActionBar;
    (ab.ios as UINavigationBar).topItem.title = "I wwaaas changed";
}
```

**Comment by Lelelo1** [comment]
@NickIliev 
I am assigning an EventHandler for `loaded` on the actionBar in navigiatingTo. It should be the same thing.

Also, the setTimout is till necessary In your example. Or it won't give effect.

**Comment by NickIliev** [comment]
@Lelelo1 yeah, it seems that I've missed the loaded handler. It seems that we still need to next VM pass so that we could get a reference to the native iOS UIBarController. Marking the issue as bug and enhancement - meanwhile, use the setTimeout workaround.
